### PR TITLE
feat: Add pattern ingester support in SSD mode

### PR DESCRIPTION
**Pull request by cyriltovena**
**What this PR does / why we need it**:

This adds an option to activate pattern ingester in helm config but also support in write component used by SSD mode.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Review by trevorwhitney — APPROVED** [review]
LGTM!

File: docs/sources/setup/install/helm/reference.md [hunk 1]
```diff
@@ -5211,7 +5211,7 @@ null
...
...p.{{ template \"loki.fullname\" $ }}-results-cache.{{ $.Release.Namespace }}.svc\n        timeout: {{ .timeout }}\n        update_interval: 1m\n  {{- end }}\n  {{- end }}\n\n{{- with .Values.loki.storage_config }}\nstorage_config:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\n{{- with .Values.loki.query_scheduler }}\nquery_scheduler:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\n{{- with .Values.loki.compactor }}\ncompactor:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\n{{- with .Values.loki.analytics }}\nanalytics:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\n{{- with .Values.loki.querier }}\nquerier:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\n{{- with .Values.loki.index_gateway }}\nindex_gateway:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\n{{- with .Values.loki.frontend }}\nfrontend:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\n{{- with .Values.loki.frontend_worker }}\nfrontend_worker:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\n{{- with .Values.loki.distributor }}\ndistributor:\n  {{- tpl (. | toYaml) $ | nindent 4 }}\n{{- end }}\n\ntracing:\n  enabled: {{ .Values.loki.tracing.enabled }}\n",
```
Comment (JStickler) [inline docs/sources/setup/install/helm/reference.md:5214]: Did you edit this file directly? The Helm reference should be updated in `production/helm/loki/reference.md.gotmlp`. There's a note to that effect at line 11 of the `docs/sources/setup/install/helm/reference.md` file, you might have missed it.  I don't see any edits to the template in your PR.

**Comment by JStickler** [comment]
Also any Helm changes (even documentation) requires that you bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

File: docs/sources/setup/install/helm/reference.md (hunk 1 above)
Comment (cyriltovena) [reply docs/sources/setup/install/helm/reference.md:5214]: Looking into it.

File: pkg/loki/loki.go [hunk 2]
```diff
@@ -695,7 +695,7 @@ func (t *Loki) setupModuleManager() error {
 		MemberlistKV:             {Server},
 
 		Read:    {QueryFrontend, Querier},
-		Write:   {Ingester, Distributor},
+		Write:   {Ingester, Distributor, PatternIngester},
```
[the remark this answers is not in the document]
Comment (cyriltovena) [reply pkg/loki/loki.go:698]: Yes correct they have their own ring, and the distributor uses that via the tee component.

File: docs/sources/setup/install/helm/reference.md (hunk 1 above)
Comment (cyriltovena) [reply docs/sources/setup/install/helm/reference.md:5214]: I did `make helm-docs`

**Pull request merged**